api/namex/resources/requests.py

```diff
@@ -577,38 +584,46 @@ def _get_next_set_from_solr(solr_query, start, rows):
         identifiers = search.get('identifiers', [])
-        nr_name = search.get('name', None)
-        state = search.get('state', [])
-
+        search_details = AffiliationInvitationSearchDetails.from_request_args(search)
+        
         # Only names and applicants are needed for this query, we want this query to be lighting fast
         # to prevent putting a load on namex-api.
         # Base query with the common identifier filter
         q = RequestDAO.query.filter(RequestDAO.nrNum.in_(identifiers))
 
+        if search_details.identifier:
+            q = q.filter(func.replace(RequestDAO.nrNum, " ", "").ilike(f'%{search_details.identifier.replace(" ", "")}%'))
         # Add the state filter if 'state' is provided
-        if state:
-            q = q.filter(RequestDAO.stateCd.in_(state))
+        if search_details.status:
+            q = q.filter(RequestDAO.stateCd.in_(search_details.status))
 
         # Add the nr_name filter if 'nr_name' is provided
-        if nr_name:
-            q = q.filter(RequestDAO.nameSearch.ilike(f'%{nr_name}%'))
+        if search_details.name:
+            q = q.filter(RequestDAO.nameSearch.ilike(f'%{search_details.name}%'))
+        
+        if search_details.type:
+            request_typecd = nr_filing_actions.get_request_type_array(search_details.type)
+            flattened_request_types = [item for sublist in request_typecd.values() for item in sublist]
+            q = q.filter(RequestDAO.requestTypeCd.in_(flattened_request_types))
 
         q = q.options(
-            lazyload('*'),
-            eagerload(RequestDAO.names).load_only(Name.state, Name.name),
-            eagerload(RequestDAO.applicants).load_only(Applicant.emailAddress, Applicant.phoneNumber),
-            load_only(
-                RequestDAO.id,
-                RequestDAO.nrNum,
-                RequestDAO.stateCd,
-                RequestDAO.requestTypeCd,
-                RequestDAO.natureBusinessInfo,
-                RequestDAO._entity_type_cd,
-                RequestDAO.expirationDate,
-                RequestDAO.consentFlag,
-                RequestDAO._request_action_cd,
-            ),
-        )
+                lazyload('*'),
+                eagerload(RequestDAO.names).load_only(Name.state, Name.name),
+                eagerload(RequestDAO.applicants).load_only(
+                    Applicant.emailAddress, Applicant.phoneNumber
+                ),
+                load_only(
+                    RequestDAO.id,
+                    RequestDAO.nrNum,
+                    RequestDAO.stateCd,
+                    RequestDAO.requestTypeCd,
+                    RequestDAO.natureBusinessInfo,
+                    RequestDAO._entity_type_cd,
+                    RequestDAO.expirationDate,
+                    RequestDAO.consentFlag,
+                    RequestDAO._request_action_cd
+                ))
+        q = q.offset((search_details.page - 1) * search_details.limit).limit(search_details.limit)
```

Comment:
This needs to be conditional, otherwise you're going to bust the current implementation by limiting it to a default?

Reply:
As discussed, easier to just set the limit to 100k for now (will cover all the NRs)